src/music.py

```diff
@@ -277,46 +334,97 @@ def articulation(self):
     def articulation(self, articulation):
         self._articulation = articulation
 
-    """You can change everything about the note except the pitch itself"""
-
     def __eq__(self, other):
         return self.duration == other.duration and self.pitch == other.pitch
 
     # Mingus interface
 
-    def is_valid_note(self, note): 
-        return notes.is_valid_note(note)
-
-    def int_to_note(self, this_int):
-        # C -> 0
-        # C# -> 1
-        # D -> 2
-        # D# -> 3
-        # E -> 4
-        return notes.int_to_note(this_int)
+    def augment(self):
+        """Raises the note by a half step"""
+        self._mingus_note = mingus_notes.augment(str(self._mingus_note))
 
-    def augment_note(self):
-        self._note = notes.augment(str(self._note))
+        if 'b' in self._accidentals:
+            self._accidentals.remove('b')
+        elif '#' in self._accidentals:
+            self._accidentals.replace('#', 'X')
+        else:
+            self._accidentals += '#'
         return True
 
-    def diminish_note(self):
-        self._note = notes.diminish(str(self._note))
-        return True
+    def diminish(self):
+        """Lowers the note by a half step"""
+        self._mingus_note = mingus_notes.diminish(str(self.mingus()))
 
-    def to_minor(self):
-        self._note = notes.to_minor(self._note)
-        return True
-    
-    def to_major(self):
-        self._note = notes.to_major(self._note)
+        if 'X' in self._accidentals:
+            self._accidentals.replace('X', '#')
+        elif '#' in self._accidentals:
+            self._accidentals.remove('#')
+        else:
+            self._accidentals += 'b'
         return True
-    
-    def transpose(self, transpose_t):
-        self._note.transpose(int(transpose_t))
+
+    def transpose(self, half_steps):
+        """Raises/Lowers the note"""
+        if half_steps < 0:
+            for _ in range(abs(half_steps)):
+                self.diminish()
+        elif half_steps > 0:
+            for _ in range(abs(half_steps)):
+                self.augment()
+
         return True
 
 
 class Rest(Note):
```

Comment:
Override Note methods that are nonsensical in Rest. They now throw exceptions when you access the pitch of a rest.